internal/eval/evalers_test.go

```diff
@@ -1234,6 +1234,28 @@ func TestContainsAnyNode(t *testing.T) {
 			})
 		}
 	}
+
+	t.Run("not quadratic", func(t *testing.T) {
```

Comment:
This test demonstrates that a worst-case search for the intersection between two sets was previously quadratic.